lib/recognizer/accounts.ex

```diff
@@ -163,7 +163,9 @@ defmodule Recognizer.Accounts do
 
   """
   def register_user(attrs, opts \\ []) do
-    if attrs["newsletter"] === "true", do: Recognizer.Hal.update_newsletter(attrs)
+    if Map.has_key?(attrs, "newsletter") do
+      if attrs["newsletter"] === "true", do: Recognizer.Hal.update_newsletter(attrs)
+    end
```

Comment:
When the user does not check the box do we sent a value or nothing? I think we can refactor this either way.

If we omit the `newsletter` key if unselected this should work because `Map.get/3` returns `nil` for no hit, which is falsey:

```suggestion
    if Map.get(attrs, "newsletter") do
       Recognizer.Hal.update_newsletter(attrs)
    end
```
One small change and we can check explicitly for `"true"`:

```suggestion
    if Map.get(attrs, "newsletter") == "true" do
      Recognizer.Hal.update_newsletter(attrs)
    end
```